crates/net/network/src/peers/manager.rs

```diff
@@ -451,7 +459,10 @@ impl PeersManager {
     ///
     /// Returns `None` if no peer is available.
     fn best_unconnected(&mut self) -> Option<(PeerId, &mut Peer)> {
-        let mut unconnected = self.peers.iter_mut().filter(|(_, peer)| peer.state.is_unconnected());
+        let mut unconnected = self.peers.iter_mut().filter(|(_, peer)| {
+            peer.state.is_unconnected() &&
+                (!self.connect_trusted_nodes_only || peer.kind.is_trusted())
+        });
```

Comment:
the way the ranking is currently implemented, trusted peers are not necessarily prioritized.

we could do this in two steps instead:

1. find unconnected peer that's not banned, if there's one return that
2. otherwise proceed with `best_unconnected` if connect_trusted_nodes_only is false

wdyt?

Reply:
Sounds good to me but, should we also prioritize between trusted nodes? If so, might be better to add the logic inside the comparison instead.

Reply:
I don't think we need to compare them and can just treat them equally, only need to check if they're currently banned

Reply:
The compiler complained about mutably borrowing the _peers_ `HashMap` more than once, so I ended up adding the check inside the loop. Also filtered banned peers beforehand, to reduce checks afterwards.